src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -1,150 +1,161 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
-using System;
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Utf16;
+using System.Threading.Tasks;
 
 namespace System.Text.Utf8
 {
-    public struct Utf8String : IEquatable<Utf8String>, IComparable<Utf8String>
+    public struct Utf8String : IEnumerable<UnicodeCodePoint>, IEquatable<Utf8String>// TODO:, IComparable<Utf8String> 
+                               // TODO: fix it when we get Span<byte> runtime support
     {
-        static readonly Utf8String s_empty = new Utf8String(new byte[0]);
```

Comment:
I started this class from a scratch while learning more about Unicode and then unified to your old version. Will revert